### PR TITLE
fix: Adding invite to app permission to fix share modal getting auto-closed issue

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

A public app that the logged-in user doesn't have access to was auto-closing the share modal when opened. This was happening because we were fetching workspace even when the user did not have the invite to app permission. This is now fixed with this PR.

#### PR fixes following issue(s)
Fixes [#26870](https://github.com/appsmithorg/appsmith/issues/26870)

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by dipyamanbiswas07 — APPROVED** [review]
LGTM

File: app/client/src/utils/hooks/useWorkspace.tsx [hunk 1]
```diff
@@ -5,16 +5,30 @@ import { getCurrentUser } from "selectors/usersSelectors";
...
+    ...userWorkspacePermissions,
+    ...userAppPermissions,
+  ]);
 
   useEffect(() => {
     if (!currentUser || currentUser.username === ANONYMOUS_USERNAME) return;
 
-    if ((!workspace || !workspace.userPermissions) && workspaceId) {
+    if (
+      (!workspace || !workspace?.userPermissions) &&
```
Comment (berzerkeer) [inline app/client/src/utils/hooks/useWorkspace.tsx:28]: Will there be a case where we have workspace but not userPermissions in workspace ?

**Review by berzerkeer — APPROVED** [review]

File: app/client/src/utils/hooks/useWorkspace.tsx (hunk 1 above)
Comment (ankitakinger) [reply app/client/src/utils/hooks/useWorkspace.tsx:28]: actually no... but just added it safe-side... its better to have checks

Comment (berzerkeer) [reply app/client/src/utils/hooks/useWorkspace.tsx:28]: Thanks!

**Comment by RakshaKShetty** [comment]
Manually tested the issue looks good

**Pull request merged**